src/types.d.ts

```diff
@@ -43,6 +43,35 @@ export interface EnterpriseLearnerData {
   staffEnterpriseCustomer: Types.EnterpriseCustomer;
 }
 
+interface DueDate {
+  name?: string;
+  date?: string;
+  url?: string;
+}
```

Comment:
[nit/curious] While not super important to get the `DueDate` types correct (i.e., since it's a no-op / unused in the UI today, where `due_dates` is always be an empty array returned by the source API), are we sure these fields are optional?

Looking at the `get_due_dates` implementation ([source](https://github.com/openedx/edx-platform/blob/9cade7a3488eee24109311cb145747a01348df14/lms/djangoapps/course_api/api.py#L245)), it seems `name` is always going to have a value, especially given the fallback to `UNKNOWN_BLOCK_DISPLAY_NAME` in the try/except.

I _think_ we'd be safe to assume these fields are not optional.